backend/amd64/proc.ml

```diff
@@ -622,6 +640,16 @@ let init () =
   end else
     num_available_registers.(0) <- 13
 
+(* This is not always the same as [all_phys_regs], as some physical registers
+   may not be allocatable (e.g. rbp when frame pointers are enabled). *)
+let precolored_regs =
+  let ints = Array.sub hard_int_reg 0 num_available_registers.(0) in
+  let floats = Array.sub hard_float_reg 0 num_available_registers.(1) in
+  let vec128s = Array.sub hard_vec128_reg 0 num_available_registers.(1) in
```

Comment:
is it possible that `precolored_regs` is computed before the first ever call to `Proc.init ()` ?
If so, `num_available_registers`  can be determined without running it, based on the configure option `fp`.

Reply:
Edited to check fp instead of relying on init